### PR TITLE
gptdocs: add option to generate docs for a list of events

### PR DESCRIPTION
**Pull request by rafaeldtinoco**
This is a fix to tracee-gptdocs (currently dead locked due to a last min addiction).

### CODE REVIEW

File: pkg/cmd/urfave/urfave.go [hunk 1]
```diff
@@ -198,10 +198,13 @@ func getContainerMode(cfg tracee.Config) printer.ContainerMode {
 	return containerMode
 }
 
-func GetGPTDocsRunner(c *cli.Context, k string, t float64, m int) (*cmd.GPTDocsRunner, error) {
+func GetGPTDocsRunner(c *cli.Context, k string, t float64, m int, e []string) (
```
Comment (geyslan) [inline pkg/cmd/urfave/urfave.go:201]: context is not being used?

File: pkg/cmd/gptdocs.go [hunk 2]
```diff
@@ -98,39 +98,71 @@ func (r GPTDocsRunner) Run(ctx context.Context) error {
...
 
+		// Check if the event is in the given list of events and skip if it not
+
+		if len(r.GivenEvents) > 0 {
+			found := false
+			for _, given := range r.GivenEvents {
+				if strings.Contains(evt.Name, given) {
+					found = true
+				}
+			}
```
Comment (geyslan) [inline pkg/cmd/gptdocs.go:151]: should break when found?

File: pkg/cmd/gptdocs.go [hunk 3]
```diff
@@ -98,39 +98,71 @@ func (r GPTDocsRunner) Run(ctx context.Context) error {
...
 				return nil
 			default:
+				logger.Debug("file already exists", "file", fileName)
 				continue
 			}
 		}
 
+		// Check if the event is in the given list of events and skip if it not
+
+		if len(r.GivenEvents) > 0 {
```
Comment (geyslan) [inline pkg/cmd/gptdocs.go:145]: It's safe to loop over slice (which can be empty), so we can get rid of this check.

**Review by geyslan — APPROVED** [review]
LGTM

File: pkg/cmd/urfave/urfave.go (hunk 1 above)
Comment (rafaeldtinoco) [reply pkg/cmd/urfave/urfave.go:201]: true, removing

File: pkg/cmd/gptdocs.go (hunk 2 above)
Comment (rafaeldtinoco) [reply pkg/cmd/gptdocs.go:151]: nope, this is correct. continue has to be outside this range loop.

File: pkg/cmd/gptdocs.go (hunk 3 above)
Comment (rafaeldtinoco) [reply pkg/cmd/gptdocs.go:145]: I want the debug and if I remove the check the debug would always trigger.

**Pull request merged**